### PR TITLE
Initial CLI code

### PR DESCRIPTION
**Pull request by stuartleeks**
- [x] Add initial code for a `tre` CLI for working with the AzureTRE API
- [x] Update shared service deployment script to use the CLI
- [x] Document the need to enable device code flow on root API and workspace APIs

Future work:
- #2706 
- #2707
- #2708
- #2709
- #2710

### CODE REVIEW

**Comment by stuartleeks** [comment]
/test

(added CLI install to CI dev container action)

File: cli/README.md [hunk 1]
```diff
@@ -0,0 +1,213 @@
+# TRE CLI
+
+**WARNING - this CLI is currently experimental**
+
+## Installation
+
+Currently, I've only tested installation in the dev container. In this context, `make install-cli` should work
```
Comment (martinpeck) [inline cli/README.md:7]: Suggest rewording this to avoid `I've`
e.g. `Currently, installation has only been tested within the dev container.`

File: cli/setup.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env python
+
+from setuptools import find_packages
+from setuptools import setup
+
+PROJECT = 'tre'
+VERSION = '0.1.1'
+
+try:
+    long_description = open('README.md', 'rt').read()  # TODO: add long description
```
Comment (martinpeck) [inline cli/setup.py:10]: what's left to "TODO"?

File: cli/tre/commands/shared_services/shared_service.py [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    shared_service_id = shared_service_context.shared_service_id
+    if shared_service_id is None:
+        raise click.UsageError('Missing shared_service ID')
+
+    client = ApiClient.get_api_client_from_config()
+    response = client.call_api(log, 'GET', f'/api/shared-services/{shared_service_id}', )
+    output(response.text, output_format=output_format, query=query, default_table_query=r"sharedServices[].{id:id,name:templateName, version:templateVersion, is_enabled:isEnabled, status: deploymentStatus}")
+
+#
+# TODO - add PATCH (and ?set-enabled)
```
Comment (martinpeck) [inline cli/tre/commands/shared_services/shared_service.py:36]: do we need these TODOs in here and, if so, should we group this one with the one below?

**Review by martinpeck — APPROVED** [review]
LTGM! Nice 🥇

File: cli/tre/commands/shared_services/shared_service.py (hunk 3 above)
Comment (stuartleeks) [reply cli/tre/commands/shared_services/shared_service.py]: Grouped with the one below as suggested

File: cli/setup.py (hunk 2 above)
Comment (stuartleeks) [reply cli/setup.py]: oops - removed

File: cli/README.md (hunk 1 above)
Comment (stuartleeks) [reply cli/README.md]: 👍

**Comment by stuartleeks** [comment]
/test-extended

re-run due to `A hash conflict was encountered for the role Assignment ID. Please use a new Guid`
